docs/high-availability/provision.sh

```diff
@@ -0,0 +1,181 @@
+    ### That is how we modulate leader.  Each node will continously either
+    ### ensure that the manifests are in this dir, or else, are in the kubelet manifest dir.
+    cp /vagrant/kube-scheduler.manifest /vagrant/kube-controller-manager.manifest /srv/kubernetes
+
+    ### All nodes will run an API Server.  This is because API Server is stateless, so its not a problem
+    ### To serve it up everywhere.
+    cp /vagrant/kube-apiserver.manifest /etc/kubernetes/manifests/
+}
+
+function write_podmaster_config {
+    touch /var/log/kube-scheduler.log
+    touch /var/log/kube-controller-manager.log
+
+    ### These DO NOT go in manifest. Instead, we mount them here.
+    ### We let podmaster swap these in and out of the manifests directory
+    ### based on its own internal HA logic.
+    cp /vagrant/kube-controller-manager.manifest /srv/kubernetes/
+    cp /vagrant/kube-scheduler.manifest /srv/kubernetes/
+
+    #### Finally, the podmaster is the mechanism for election
+    cp /vagrant/podmaster.json /etc/kubernetes/manifests/
+}
+
+function poll {
+    ### wait 10 minutes for kube-apiserver to come online
+    for i in `seq 1 600`
+    do
+           sleep 2
+            echo $i
+           ### Just testing that the front end comes up.  Not sure how to test total entries etc... (yet)
+           ( curl "localhost:8080" > result || echo "failed on attempt $i, retrying again.. api not up yet. " )
+           ( cat result || echo "no result" )
+           if ( cat result | grep -q api ) ; then
+                break
+           else
+              echo "continue.."
+           fi
+   done
+   if [ $i == 600 ]; then
+        exit 2
+   fi
+}
+
+function install_components {
+    ### etcd node - this node only runs etcd in a kubelet, no flannel.
+    ### we dont want circular dependency of docker -> flannel -> etcd -> docker
+    if [ "`hostname`" == "kube0.ha" ]; then
+            write_etcd_manifest
+            start_kubelet
+
+            ### precaution to make sure etcd is writable, flush iptables.
+             iptables -F
+        ### minions: these will each run their own api server.
+    else
+        ### Make sure etcd running, flannel needs it.
+        test_etcd
+        start_kubelet
+
+        ### Flannel setup...
+        ### This will restart the kubelet and docker and so on...
+        /vagrant/provision-flannel.sh
+
+        echo "Now pulling down flannel nodes. "
+        curl -L http://kube0.ha:2379/v2/keys/coreos.com/network/subnets | python -mjson.tool
+
+        echo " Inspect the above lines carefully ^."
+        ### All nodes run api server
+        write_api_server_config
+
+        ### controller-manager will turn on and off
+        ### and same for kube-scheduler
+        write_podmaster_config
+
+        # finally, for us to creaet public ips for k8petstore etc, we need the proxy running.
+         service kube-proxy start
+         service kube-proxy status
+    fi
+}
+
+initialize
+install_components
+iptables -F
```

Comment:
this is something i do to make sure the kube-proxy works properly.    without it there can be some old lingering rejected packets for some reason.